File: dotnet/ThirtyFour.UI/Window/BrowserWindow.cs
using OpenQA.Selenium;

namespace ThirtyFour.UI.Window
{
    /// <summary>
    /// The BrowserWindow class provides an easy method for finding and switching to additional browser windows
    /// </summary>
    public class BrowserWindow
    {
        const double DEFAULT_TIMEOUT_IN_SECONDS = 10;

        /// <summary>
        /// The driver instance
        /// </summary>
        public IWebDriver Driver { get; private set; }

        /// <summary>
        /// The default window handle
        /// </summary>
        public string DefaultHandle { get; private set; }

        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="driver">The IWebDriver instance</param>
        public BrowserWindow(IWebDriver driver) : this(driver, new DefaultMatcher(), DEFAULT_TIMEOUT_IN_SECONDS)
        {
        }

        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="driver">The IWebDriver instance</param>
        /// <param name="matcher">An instance of IMatcher</param>
        /// <param name="timeoutInSeconds">The number of seconds before timing out</param>
        public BrowserWindow(IWebDriver driver, IMatcher matcher, double timeoutInSeconds)
        {
            this.DefaultHandle = driver.CurrentWindowHandle;
            this.Driver = driver;
            matcher.MatchWindow(driver, timeoutInSeconds);
        }

    }
}

File: java/src/main/java/com/github/danhumphrey/thirtyfour/ui/window/BrowserWindow.java
package com.github.danhumphrey.thirtyfour.ui.window;

import org.openqa.selenium.WebDriver;

/**
 * The BrowserWindow class provides an easy method for finding and switching to additional browser windows
 *
 */
public class BrowserWindow {

	private WebDriver driver;
	private String defaultHandle;
	
	final static long DEFAULT_TIMEOUT_IN_SECONDS = 10;

	
	/**
	 * Constructor
	 * 
	 * @param driver the <code>WebDriver</code> instance
	 * @throws Exception 
	 */
	public BrowserWindow(WebDriver driver) throws Exception {
		this(driver, new DefaultMatcher(), BrowserWindow.DEFAULT_TIMEOUT_IN_SECONDS);
	}
	
	/**
	 * Constructor
	 * 
	 * @param driver the <code>WebDriver</code> instance
	 * @param matcher a <Matcher> instance to match the window
	 * @param timeoutInMilliseconds The number of milliseconds before timing out
	 * @throws Exception if window cannot be found
	 */ 
	public BrowserWindow(WebDriver driver, Matcher matcher, long timeoutInMilliseconds) throws Exception {
		this.driver = driver;
		this.defaultHandle = driver.getWindowHandle();
		matcher.matchWindow(driver, timeoutInMilliseconds);
	}
	
	/**
	 * @return the driver
	 */
	public WebDriver getDriver() {
		return driver;
	}
	
	/**
	 * @return the default window handle
	 */
	public String getDefaultHandle() {
		return defaultHandle;
	}

}
